frontend/src/App.tsx

```diff
@@ -35,61 +38,83 @@ const App = (): React.ReactElement => {
     currentUser,
   );
 
+  const [volunteerUser, setVolunteerUser] = useState<AuthenticatedVolunteer>(
+    null,
+  );
+
+  useEffect(() => {
+    async function getVolunteerUser() {
+      if (authenticatedUser && authenticatedUser.role === Role.VOLUNTEER) {
+        const volunteer = await volunteerAPIClient.getVolunteerByUserId(
+          authenticatedUser.id,
+        );
+        setVolunteerUser(volunteer);
+      }
+    }
+    getVolunteerUser();
+  }, []);
   return (
     <ChakraProvider theme={customTheme}>
       <AuthContext.Provider value={{ authenticatedUser, setAuthenticatedUser }}>
-        <Router>
-          <FeedbackBanner />
-          <Header />
-          <Switch>
-            <Route exact path={Routes.LOGIN_PAGE} component={Login} />
-            <Route exact path={Routes.SIGNUP_PAGE} component={Signup} />
-            <Route exact path={Routes.LANDING_PAGE} component={Home} />
-            <PrivateRoute
-              exact
-              path={Routes.USER_MANAGEMENT_PAGE}
-              component={UserManagement}
-            />
-            <PrivateRoute
-              exact
-              path={Routes.VOLUNTEER_SHIFTS_PAGE}
-              component={VolunteerShiftsPage}
-            />
-            <PrivateRoute
-              exact
-              path={Routes.ACCOUNT_PAGE}
-              component={Account}
-            />
-            <Route
-              exact
-              path={Routes.VIEW_DONATIONS}
-              component={ViewDonationsPage}
-            />
-            <PrivateRoute
-              exact
-              path={Routes.DASHBOARD_SCHEDULE_EDIT_PAGE}
-              component={EditDashboardSchedulePage}
-            />
-            <PrivateRoute
-              exact
-              path={Routes.DASHBOARD_PAGE}
-              component={Dashboard}
-            />
-            <PrivateRoute
-              exact
-              path={Routes.SCHEDULING_PAGE}
-              component={Scheduling as React.FC}
-            />
-            <Route path={Routes.ACTION} component={Action} />
-            <Route
-              exact
-              path={Routes.FORGET_PASSWORD}
-              component={ResetPassword}
-            />
-            <Route exact path="*" component={NotFound} />
-          </Switch>
-          <Footer />
-        </Router>
+        <VolunteerContext.Provider
```

Comment:
This will create a context no matter if they are a volunteer or donor which is probably not what we want? I wasn't sure how to create it exclusively if they were a volunteer. I think the provider has to be on this level because we want them to be accessible to the entire application.

Reply:
putting the context provider here is fine, we should also have a volunteer context dispatcher here:

https://github.com/uwblueprint/starter-code-v2/blob/e48f2e4b6288a417e6029522fd6934a4d50d25e1/frontend/src/App.tsx#L39